test/widgets/compose_box_test.dart

```diff
@@ -1354,4 +1400,462 @@ void main() {
+      if (shouldContinue) {
+        await tester.tap(find.byWidget(actionButton));
+      } else {
+        await tester.tap(find.byWidget(cancelButton));
+      }
+    }
+
+    // Test the "Discard…?" confirmation dialog when you tap "Edit message" in
+    // the action sheet but there's text in the compose box for a new message.
+    void doTestInterruptComposingFromActionSheet({required Narrow narrow}) {
+      testWidgets('interrupting new-message compose: $narrow', (tester) async {
+        TypingNotifier.debugEnable = false;
+        addTearDown(TypingNotifier.debugReset);
+
+        final messageId = msgIdInNarrow(narrow);
+        await prepareEditMessage(tester, narrow: narrow);
+        checkNotInEditingMode(tester, narrow: narrow);
+
+        await enterContent(tester, 'composing new message');
+
+        // Expect confirmation dialog; tap Cancel
+        await startInteractionFromActionSheet(tester, messageId: messageId);
+        await expectAndHandleDiscardConfirmation(tester, shouldContinue: false);
+        check(connection.takeRequests()).isEmpty();
+        // fetch-raw-content request wasn't actually sent;
+        // take back its prepared response
+        connection.clearPreparedResponses();
+
+        // Twiddle the input to make sure it still works
+        checkNotInEditingMode(tester,
+          narrow: narrow, expectedContentText: 'composing new message');
+        await enterContent(tester, 'composing new message…');
+        checkContentInputValue(tester, 'composing new message…');
+
+        // Try again, but this time tap Discard and expect to enter an edit session
+        await startInteractionFromActionSheet(tester,
+          messageId: messageId, originalRawContent: 'foo');
+        await expectAndHandleDiscardConfirmation(tester, shouldContinue: true);
+        await tester.pump();
+        await checkAwaitingRawMessageContent(tester);
+        await tester.pump(Duration(seconds: 1)); // fetch-raw-content request
+        check(connection.takeRequests()).length.equals(1);
+        checkContentInputValue(tester, 'foo');
+        await enterContent(tester, 'bar');
+
+        // Save; check that the request is made and the compose box resets.
+        connection.prepare(json: UpdateMessageResult().toJson());
+        await tester.tap(find.widgetWithText(ZulipWebUiKitButton, 'Save'));
+        checkRequest(messageId, prevContent: 'foo', content: 'bar');
+        await tester.pump(Duration.zero);
+        checkNotInEditingMode(tester, narrow: narrow);
+      });
+    }
+    doTestInterruptComposingFromActionSheet(narrow: channelNarrow);
+    doTestInterruptComposingFromActionSheet(narrow: topicNarrow);
+    doTestInterruptComposingFromActionSheet(narrow: dmNarrow);
+
+    // Test the "Discard…?" confirmation dialog when there's text in the compose
+    // box for a new message, but you've said you want to edit a message.
+    void doTestInterruptComposingFromFailedEdit({required Narrow narrow}) {
```

Comment:
This comment doesn't seem to match the name. (The comment seems to say the same thing as the comment on doTestInterruptComposingFromActionSheet above.)